src/main/java/bio/terra/service/resourcemanagement/flight/AzureStorageMonitoringProvider.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.resourcemanagement.flight;
+
+import bio.terra.service.resourcemanagement.azure.AzureMonitoringService;
+import bio.terra.stairway.RetryRule;
+import bio.terra.stairway.RetryRuleNone;
+import bio.terra.stairway.Step;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AzureStorageMonitoringProvider {
+
+  private final AzureMonitoringService monitoringService;
+
+  public AzureStorageMonitoringProvider(AzureMonitoringService monitoringService) {
+    this.monitoringService = monitoringService;
+  }
+
+  /**
+   * Configure the steps needed to configure storage account logging and monitoring. If the dataset
+   * or snapshot has secure monitoring enabled, this will configure extra log retention and
+   * monitoring resources
+   *
+   * @param isSecureMonitoringEnabled Is this a dataset or snapshot with secure monitoring enabled?
+   */
+  public List<StepDef> configureSteps(boolean isSecureMonitoringEnabled) {
+    List<StepDef> steps = new ArrayList<>();
+    RetryRuleNone noRetry = RetryRuleNone.getRetryRuleNone();
```

Comment:
Ah we don't!  I'd started down the road having this class actually add the steps itself but it was - I think - needlessly complicated (you need to pass in a consumer to call the addStep from the scope of the flight...just kind of unwieldy

Reply:
Oh wait, sorry, now I remember.  We do actually because step doesn't contain its retry settings so we need an object to return that contains both (in order to handle the case where need an actual retry)